lib/tag/vdom.js

```diff
@@ -38,10 +38,11 @@ function mountTo(root, tagName, opts) {
 
 }
 
-riot.tag = function(name, html, css, fn) {
+riot.tag = function(name, html, css, attrs, fn) {
+  if (typeof attrs == 'function') {fn = attrs; attrs=css}
```

Comment:
Thanks, although your tag causes stack overflow, `<my-tag>` in the html part is recursive on itself, but once I figured that out, I think I have it.  The compiler skips css if not provided, so the attrs needs to look there.

```javascript
riot.tag = function(name, html, css, attrs, fn) {
  // if css test as attrs, then it is, else set attrs to '' for backward compatibility 
  if (typeof attrs == 'function') {fn = attrs; attrs = /^[\w\-]+\s?=/.test(css) ? css : ''} 
  if (typeof css == 'function') fn = css
  else if (css) injectStyle(css)
  tagImpl[name] = { name: name, tmpl: html, attrs: attrs, fn: fn }
  return name
}
```

[this](http://plnkr.co/edit/saGdM0jLwAp36RVYMLTI?p=preview) works and all my tests pass.  I will re-submit soon.  I need to merge in #670 before I can get the test to work with your sample tag,